model_analyzer/config/run/run_config_generator.py

```diff
@@ -135,7 +137,35 @@ def generate_run_config_for_model_sweep(self, model, model_sweep):
                     RunConfig(model.model_name(), model_config, perf_config,
                               model.triton_server_environment()))
 
+    def _apply_model_config_defaults(self, config_dict, cpu_only):
+        if "instance_group" not in config_dict:
+            config_dict["instance_group"] = [{}]
+        for instance_group in config_dict["instance_group"]:
+            self._apply_instance_group_defaults(instance_group, cpu_only)
+
+    def _apply_instance_group_defaults(self, instance_group, cpu_only):
+        if "count" not in instance_group:
+            instance_group["count"] = 1
+        if "kind" not in instance_group:
+            if (cpu_only):
+                instance_group["kind"] = "KIND_CPU"
+            else:
+                instance_group["kind"] = "KIND_GPU"
+
     def generate_model_config_combinations(self, value):
+        configs = self._generate_model_config_combinations_helper(value)
+        if not self._default_config_in_configs(configs):
+            self._add_default_config(configs)
+        return configs
+
+    def _default_config_in_configs(self, configs):
+        return None in configs
```

Comment:
Can we move this to L156?

Reply:
This is intentional and self-documenting